angelsrefining/prototypes/angels-functions.lua

```diff
@@ -1375,17 +1375,54 @@ function angelsmods.functions.remove_crafting_category(crafting_machine_type, cr
   end
 end
 
+-------------------------------------------------------------------------------
+-- MODIFY FAST_REPLACE_CATEGORY -----------------------------------------------
+-------------------------------------------------------------------------------
+function angelsmods.functions.set_fast_replace_category(crafting_machine_type, crafting_machine_name, next_upgrade)
+  --search for new category (if needed), skip if identical
+  if not data.raw[crafting_machine_type] then return end
+  local crafting_machine1 = data.raw[crafting_machine_type][crafting_machine_name]
+  local crafting_machine2 = data.raw[crafting_machine_type][next_upgrade]
+  if not crafting_machine1 then return end
+  if not crafting_machine2 then return end
+  --get current FRC
+  local FRC1 = crafting_machine1.fast_replaceable_group
+  local FRC2 = crafting_machine2.fast_replaceable_group
+
+  if FRC1 == FRC2 then
+    --no need to change
+  elseif FRC2 ~= nil then --change it
+    crafting_machine1.fast_replaceable_group = FRC2
+    if crafting_machine1.collision_box[1][1] ~= crafting_machine2.collision_box[1][1] then
+      if crafting_machine1.collision_box[2][2] ~= crafting_machine2.collision_box[2][2] then
+        --boxes don't match... nil out the properties
+        crafting_machine1.fast_replaceable_group = nil
+        --clobber next_upgrade too
+        crafting_machine1.next_upgrade = nil
+      else --set fast_replace group
+        if FRC ~= nil then
+          crafting_machine1.fast_replaceable_group = FRC2

```

Comment:
Line 1395 you set machine 1 already to FRC2 and later on line 1404 you conditionally set it again?

Reply:
this is why i should not code while tired... swapping methods mid idea tends to get messy